File: golang-backend/internal/handlers/projectpermissionhandler/dto.go
package projectpermissionhandler

import (
	"github.com/aportela/doneo/internal/handlers/rolehandler"
	"github.com/aportela/doneo/internal/handlers/userhandler"
)

// TODO: share userbase with user handler ??
type userBase struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}

// TODO: share rolebase with role handler ??
type roleBase struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}

type addRequest struct {
	User userBase `json:"user"`
	Role roleBase `json:"role"`
}

type ProjectPermissionResponse struct {
	ID   string                       `json:"id"`
	User userhandler.UserBaseResponse `json:"user"`
	Role rolehandler.RoleResponse     `json:"role"`
}

type searchResponse struct {
	ProjectPermissions []ProjectPermissionResponse `json:"projectPermissions"`
}

File: vite-vue-ts-frontend/src/modules/project-permissions/types/dto.ts
import type { UserBaseResponse } from "../../users/types/dto";
import type { RoleResponse } from "../../roles/types/dto";

export type AddRequest = {
  user: UserBaseResponse;
  role: RoleResponse;
};

export type ProjectPermissionResponse = {
  id: string;
  user: UserBaseResponse;
  role: RoleResponse;
};

export type SearchResponse = {
  projectAttachments: ProjectPermissionResponse[];
};
